include/numerics/type_vector.h

```diff
@@ -47,6 +53,24 @@ template <typename T, typename D, bool asd>
 MetaPhysicL::DualNumber<T, D, asd> abs(const MetaPhysicL::DualNumber<T, D, asd> & in);
 template <typename T, typename D, bool asd>
 MetaPhysicL::DualNumber<T, D, asd> abs(MetaPhysicL::DualNumber<T, D, asd> && in);
+
+template <typename T, typename D, bool asd>
+using ADRealEigenVector = Eigen::Matrix<MetaPhysicL::DualNumber<T, D, asd>, Eigen::Dynamic, 1>;
```

Comment:
Shouldn't this also go inside the LIBMESH_HAVE_EIGEN ifdef?

Reply:
yes